### PR TITLE
feat(e2e): add e2e cases for br with x-k8s tidbcluster

### PR DESCRIPTION
**Pull request by just1900**
### What problem does this PR solve?

Add e2e cases for br with x-k8s tidbcluster.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Add e2e cases for br with x-k8s tidbcluster.
```

### CODE REVIEW

File: tests/e2e/br/br.go [hunk 1]
```diff
@@ -326,6 +347,143 @@ func createTidbCluster(f *e2eframework.Framework, name string, version string, e
...
+		tc1.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+
+		tc2.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+		tc2.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+
+		tc3.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+		tc3.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+	}
+
+	ginkgo.By("Creating tidb clusters with TLS enabled")
```
Comment (DanielZhangQD) [inline tests/e2e/br/br.go:437]: It will execute this code no matter TLS is enabled or not.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by KanShiori — APPROVED** [review]

**Pull request merged**